### DISCUSSION TITLE
멀티모드 텔레포트 동기화 안되는 현상

### DISCUSSION THREAD

**Discussion by jgw1413** [naverz/zepeto-studio-kor#1147] [5. Q&A]
현재 멀티 월드 개발중인데 인터페이스 써서 클라이언트 동기화를 구현하고 있습니다. 
서버에 브로드캐스트해서 클라이언트가 메세지 받는것까지 잘 확인 됐습니다.

문제는 제스쳐는 잘 되는데 텔레포트가 잘 안됩니다.... 텔레포트 지점으로 순간이동이 아니라 뛰어(Move)갑니다.

전에 했던 질문에서 어떤분께서 이동동기화를 꺼주셔야 된다고 했는데 끄면 텔레포트 했을 때 클라이언트 캐릭터가 아예 움직이질 안더라구요...

현재 제페토에서 제공하는 멀티 샘플씬 이용해서 만들고 있는데 혹시 이런 이슈 겪으신분들께 도움 정말 부탁드립니다!

**Comment by Sonak0930** [comment]
저도 정확하게 같은 문제를 겪고 있는데 해결방법을 못찾았습니다. Multi Moving block example에서도 같은 현상이 일어나더라구요.

**Reply by jgw1413** [reply]
멀티 무빙블럭 샘플에서 추락했을때 서로 어떻게 텔레포트 처리하는지 궁금해서 봤는데 코드에서 힌트 얻고 해결했습니다 감사합니다 ^_^

**Reply by Sonak0930** [reply]
혹시 어떻게 해결하셨나요? 무빙블럭에서 로컬플레이어가 떨어지면 다시 안올라오던거 같아서요

**Comment by hyunjoon-park-metaspace** [comment]
텔레포트 패킷을 받았을 때

1. 해당 플레이어에 `character.enabled = false`를 수행해 캐릭터 컨트롤러 강제 중지
2. `character.transform.SetPositionAndRotation(pos, rot)`을 수행해 위치와 회전 적용
3. `character.enabled = true`를 수행해 캐릭터 컨트롤러 재활성

을 하시면 일반적으로 해결됩니다.

**Reply by Sonak0930** [reply]
알려주신 방법으로 해봤는데 어떨 때는 되고, 어떨 때는 뛰어가네요. 텔레포트 직후에 시간 텀을 두고 천천히 들어가면 동기화가 되는 거 같은데, 연속으로 텔레포트하면 이전처럼 뛰어가는 거 같아요. 혹시 이런 문제 겪으셨나요?

**Reply by jgw1413** [reply]
저는 올바른 방법은 아니지만 클라이언트 캐릭터 업데이트 부분에서 로컬거리와 보여지는 클라이언트 거리가 일정거리 이상일때 같은위치로 이동하도록 제어문을 추가했어요
멀티부빙블록 샘플도 이런 방식으로 구현이 되더라구요
